File: app/src/main/java/fergaral/datetophoto/activities/StoragePermissionDeniedActivity.java
package fergaral.datetophoto.activities;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.provider.Settings;
import android.support.annotation.NonNull;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import fergaral.datetophoto.R;
import fergaral.datetophoto.utils.Utils;

public class StoragePermissionDeniedActivity extends StoragePermissionBaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_storage_permission_denied);

        Button permissionBtn = (Button) findViewById(R.id.permissionBtn);

        if(permissionBtn != null) {
            permissionBtn.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    requestPermissionClicked(v);
                }
            });
        }
    }


    @Override
    public void onBackPressed() {
        finish(); //Since it's the first screen of the app, we simply leave it
    }

    @Override
    public void onPermissionGranted() {
        startActivity(new Intent(this, PhotosActivity.class));
    }
}

File: app/src/main/java/fergaral/datetophoto/activities/StoragePermissionDeniedActivity.kt
package fergaral.datetophoto.activities

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.core.view.ViewCompat
import androidx.core.view.updateLayoutParams
import androidx.core.view.updatePadding

import fergaral.datetophoto.R

class StoragePermissionDeniedActivity : StoragePermissionBaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_storage_permission_denied)

        val container = findViewById<View>(R.id.container)
        container.systemUiVisibility = View.SYSTEM_UI_FLAG_LAYOUT_STABLE or
                View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
        handleContainerInsets(container)
        val permissionBtn = findViewById<View>(R.id.permissionBtn) as Button
        handlePermissionButtonInsets(permissionBtn)
        permissionBtn.setOnClickListener { requestPermissionClicked() }
    }

    private fun handleContainerInsets(container: View) {
        ViewCompat.setOnApplyWindowInsetsListener(container) { v, insets ->
            v.updatePadding(
                    top = insets.systemWindowInsetTop,
                    left = insets.systemWindowInsetLeft,
                    right = insets.systemWindowInsetRight
            )
            insets
        }
    }

    private fun handlePermissionButtonInsets(permissionButton: Button) {
        ViewCompat.setOnApplyWindowInsetsListener(permissionButton) { v, insets ->
            v.updateLayoutParams<ViewGroup.MarginLayoutParams> {
                bottomMargin = insets.systemWindowInsetBottom
            }
            insets
        }
    }

    override fun onBackPressed() {
        finish() //Since it's the first screen of the app, we simply leave it
    }

    override fun onPermissionGranted() {
        startActivity(Intent(this, PhotosActivity::class.java))
    }
}
